### PR TITLE
Docstring fix

### PR DESCRIPTION
**Pull request by kmarsteller**

### CODE REVIEW

**Comment by kmarsteller** [comment]
*Reduce code duplication in docstrings of inherited classes
*Remove attributes that are still documented, that no longer exist
*Add attributes to docstring that were added but never documented.

**Comment by kmarsteller** [comment]
Hmmm getting  numerical fails on a couple py3.x tests, perhaps one of my __init__ adds is erroneous. Will attempt to debug on py3 locally.

File: openmdao/components/bsplines_comp.py [hunk 1]
```diff
@@ -110,6 +110,12 @@ def get_bspline_mtx(num_cp, num_pt, order=4, distribution='sine'):
 class BsplinesComp(ExplicitComponent):
     """
     Simple B-spline component for interpolation.
+
+    Attributes
+    ----------
+    cite : str
+        Listing of relevant citataions that should be referenced when publishing
```
Comment (swryan) [inline openmdao/components/bsplines_comp.py]: sp: citataions

File: openmdao/components/ks_comp.py [hunk 2]
```diff
@@ -81,6 +81,12 @@ class KSComp(ExplicitComponent):
     Component that aggregates a number of functions to a single value via the
     Kreisselmeier-Steinhauser Function. This new constraint is satisfied when it
     is less than or equal to zero.
+
+    Attributes
+    ----------
+    cite : str
+        Listing of relevant citataions that should be referenced when publishing
```
Comment (swryan) [inline openmdao/components/ks_comp.py]: sp: citataions

File: openmdao/drivers/pyoptsparse_driver.py [hunk 3]
```diff
@@ -79,24 +79,15 @@ class pyOptSparseDriver(Driver):
...
         List of design variables.
-    _objs : dict
-        Contains all objective info.
     _quantities : list
         Contains the objectives plus nonlinear constraints.
-    _responses : dict
-        Contains all response info.
+    cite : str
+        Listing of relevant citations that should be referenced when publishing
+        work that uses this class.
```
Comment (swryan) [inline openmdao/drivers/pyoptsparse_driver.py]: I think we did not want to duplicate for attributes that are on the base class?

File: openmdao/drivers/scipy_optimizer.py [hunk 4]
```diff
@@ -68,20 +68,17 @@ class ScipyOptimizeDriver(Driver):
...
-    _objs : dict
-        Contains all objective info.
     _exc_info : 3 item tuple
         Storage for exception and traceback information.
     _obj_and_nlcons : list
         List of objective + nonlinear constraints. Used to compute total derivatives
         for all except linear constraints.
+    cite : str
+        Listing of relevant citations that should be referenced when publishing
+        work that uses this class.
```
Comment (swryan) [inline openmdao/drivers/scipy_optimizer.py]: I think we did not want to duplicate for attributes that are on the base class?

File: openmdao/components/bsplines_comp.py (hunk 1 above)
Comment (kmarsteller) [reply openmdao/components/bsplines_comp.py]: damn copy/paste of some ridiculous speller

Comment (kmarsteller) [reply openmdao/components/bsplines_comp.py]: ✅

File: openmdao/components/ks_comp.py (hunk 2 above)
Comment (kmarsteller) [reply openmdao/components/ks_comp.py]: ✅

File: openmdao/drivers/pyoptsparse_driver.py (hunk 3 above)
Comment (kmarsteller) [reply openmdao/drivers/pyoptsparse_driver.py]: ✅

File: openmdao/drivers/scipy_optimizer.py (hunk 4 above)
Comment (kmarsteller) [reply openmdao/drivers/scipy_optimizer.py]: ✅

File: openmdao/solvers/nonlinear/broyden.py [hunk 5]
```diff
@@ -33,6 +33,9 @@ class BroydenSolver(NonlinearSolver):
 
     Attributes
     ----------
+    cite : str
```
Comment (kmarsteller) [inline openmdao/solvers/nonlinear/broyden.py]: Whoops, this is also defined in solver.py. Will remove this one, also.

**Pull request merged**